base_layer/core/src/base_node/sync/header_sync/synchronizer.rs

```diff
@@ -421,7 +421,7 @@ impl<'a, B: BlockchainBackend + 'static> HeaderSynchronizer<'a, B> {
                 },
             };
 
-            let steps_back = resp.fork_hash_index + offset as u64;
+            let steps_back = resp.fork_hash_index.saturating_add(offset as u64);
```

Comment:
This should be a protocol violation if fork_hash_index is greater than the number if block hashes sent.

Reply:
Then once that is checked, checked_add should be used - if that fails it can only be a bug in the code really (unless the chain is at height u64::MAX - NUM_CHAIN_SPLIT_HEADERS)

Reply:
That is checked.
Just a few lines later in the code. This just ensures we don't crash before we sanity checks this
https://github.com/tari-project/tari/blob/385022ad32ea10ad6e73ac6eb972f4ac90d43a4e/base_layer/core/src/base_node/sync/header_sync/synchronizer.rs#L467C5-L467C5

Reply:
maybe we should use checked_add and ban if it overflows

Reply:
I moved both bans, to in front, so technically it can't overflow anymore, but going to leave that as is.